src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/IndexOfAnyValues.cs

```diff
@@ -0,0 +1,280 @@
+        }
+
+        if (TryGetSingleRange(values, out char maxInclusive, out IndexOfAnyValues<char>? charRange))
+        {
+            return charRange;
+        }
+
+        if (values.Length == 2)
+        {
+            return new IndexOfAny2Values<char, short>(shortValues);
+        }
+
+        // IndexOfAnyAsciiSearcher for chars is faster than IndexOfAny3Values
+        if (IndexOfAnyAsciiSearcher.IsVectorizationSupported && maxInclusive < 128)
+        {
+            IndexOfAnyAsciiSearcher.ComputeBitmap(values, out Vector128<byte> bitmap, out BitVector256 lookup);
+
+            if (bitmap == AsciiLetter.Bitmap) return IndexOfAnyCharInAsciiSet<AsciiLetter>.Instance;
+            if (bitmap == AsciiLetterOrDigit.Bitmap) return IndexOfAnyCharInAsciiSet<AsciiLetterOrDigit>.Instance;
+            if (bitmap == AsciiHexDigit.Bitmap) return IndexOfAnyCharInAsciiSet<AsciiHexDigit>.Instance;
+            if (bitmap == AsciiHexDigitLower.Bitmap) return IndexOfAnyCharInAsciiSet<AsciiHexDigitLower>.Instance;
+            if (bitmap == AsciiHexDigitUpper.Bitmap) return IndexOfAnyCharInAsciiSet<AsciiHexDigitUpper>.Instance;
+
+            return Sse3.IsSupported && lookup.Contains(0)
+                ? new IndexOfAnyAsciiCharValues<IndexOfAnyAsciiSearcher.Ssse3HandleZeroInNeedle>(bitmap, lookup)
+                : new IndexOfAnyAsciiCharValues<IndexOfAnyAsciiSearcher.Default>(bitmap, lookup);
+        }
+
+        if (values.Length <= 5)
+        {
+            return values.Length switch
+            {
+                3 => new IndexOfAny3Values<char, short>(shortValues),
+                4 => new IndexOfAny4Values<char, short>(shortValues),
+                5 => new IndexOfAny5Values<char, short>(shortValues),
+                _ => throw new UnreachableException()
+            };
+        }
+
+        if (maxInclusive < 256)
+        {
+            // This will also match ASCII values when IndexOfAnyAsciiSearcher is not supported
+            return new IndexOfAnyLatin1CharValues(values);
+        }
+
+        return values.Length < Vector128<short>.Count
+            ? new IndexOfAnyCharValuesProbabilistic<ShortLoopContains>(values)
+            : new IndexOfAnyCharValuesProbabilistic<StringContains>(values);
+    }
+
+    private static bool TryGetSingleRange<T>(ReadOnlySpan<T> values, out T maxInclusive, [NotNullWhen(true)] out IndexOfAnyValues<T>? singleRangeValues)
+        where T : struct, INumber<T>, IMinMaxValue<T>
+    {
+        singleRangeValues = null;
+
+        T min = T.MaxValue;
+        T max = T.MinValue;
+
+        foreach (T value in values)
+        {
+            min = T.Min(min, value);
+            max = T.Max(max, value);
+        }
+
+        maxInclusive = max;
+
+        int range = int.CreateChecked(max - min) + 1;
+        if (range > values.Length)
+        {
+            return false;
+        }
+
+        var uniqueValues = new HashSet<T>(range);
```

Comment:
I changed this to `range < 256 ? stackalloc bool[] : new bool[]` instead of `HashSet`.
I don't think it's worth complicating the code to make use of bits instead of bools.

Reply:
Thanks.